pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -582,6 +582,14 @@ The max allowed delay for delayed delivery (in milliseconds). If the broker rece
     )
     private String configurationMetadataSyncEventTopic = null;
 
+    @FieldContext(
+            dynamic = false,
+            category = CATEGORY_SERVER,
+            doc = "If you want to enable or disable the metadata synchronizer dynamically, this value should be true."
+                + "Enabled: Pulsar will initialize itself to update the metadata synchronizer dynamically."
+    )
+    private boolean mayEnableMetadataSynchronizer = false;
```

Comment:
Maybe we can name the configuration as `forceUseSeparatedConfigurationStore`? Then we can use `isConfigurationStoreSeparated` to check the condition.
```java
    public boolean isConfigurationStoreSeparated() {
        return !Objects.equals(getConfigurationMetadataStoreUrl(), getMetadataStoreUrl()) || forceUseSeparatedConfigurationStore;
    }
```

Reply:
- Renamed `mayEnableMetadataSynchronizer` to `forceUseSeparatedConfigurationStoreInMemory`
- I did not change the method `isConfigurationStoreSeparated`, because the original behavior of this method is whether the URLs is different between Configuration Metadata Store and Local Metadata Store, we should not to change it.